File: programmers/level2/피보나치_수.java
public class 피보나치_수 {
	static
	class Solution {
		  public int solution(int n) {
		      int a = 0, b = 1;
		      int temp;
		      for(int i = 1; i < n; ++i) {
		    	temp = b;
		    	b = (b+a)%1234567;
		    	a = temp;
		      }
		      return b;
		  }
		}
	public static void main(String[] args) {
		int n = 3;
		Solution sol = new Solution();
		System.out.println(sol.solution(n));
	}
}

File: programmers/js/new/level2/피보나치_수.js
function solution(n) {
    var answer = 0;
    const dp = Array(n + 1);
    dp[0] = 0;
    dp[1] = 1;
    for (let i = 2; i <= n; ++i) {
        dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
    }
    return dp[n];
}
